### DISCUSSION TITLE
How to implement Xstream Codes support.

### DISCUSSION THREAD

**Discussion by tarrazuapps** [iptv-org/community#74] [Q&A] [answered]
This question is not related to the project, but since there are so many knowledgeable folks here, maybe you can give me a clue. I hope you don't mind little tangent.

How should a client implement Xstream Codes/ Xstream authentication and list retrieval? It seems like is just a matter to add a bunch of parameters in the get request to the server, but I am just assuming. Is there an "official" specification somewhere?

Thanks.

**Comment by tarrazuapps** [comment, marked as answer]
Found it. https://xtream-ui.org/api-xtreamui-xtreamcode/

**Comment by BellezaEmporium** [comment]
for auth and list retrieval, play with the username and password login, and use player_api or panel_api entrypoints to get channel/account data.
